File: services/customers-service/src/main/java/br/com/deltaglobalbank/customers/domain/customer/valueobjects/MaritalStatus.java
package br.com.deltaglobalbank.customers.domain.customer.valueobjects;

import java.util.Arrays;
import java.util.Locale;

public enum MaritalStatus {
    SINGLE,
    MARRIED,
    DIVORCED,
    WIDOWED,
    STABLE_UNION;

    public String toDatabaseValue() {
        return name().toLowerCase(Locale.ROOT);
    }

    public static MaritalStatus fromDatabaseValue(String value) {
        return Arrays.stream(values())
            .filter(status -> status.name().equalsIgnoreCase(value))
            .findFirst()
            .orElseThrow(() -> new IllegalArgumentException("Marital status invalid: " + value));
    }
}

File: services/customers-service/src/main/kotlin/br/com/deltaglobalbank/customers/domain/customer/valueobjects/MaritalStatus.kt
package br.com.deltaglobalbank.customers.domain.customer.valueobjects

enum class MaritalStatus {
    SINGLE,
    MARRIED,
    DIVORCED,
    WIDOWED,
    STABLE_UNION;

    fun toDatabaseValue() : String = name.lowercase()

    companion object {
        fun fromDatabaseValue(value: String): MaritalStatus =
            MaritalStatus.entries.firstOrNull { it.name.equals(value, ignoreCase = true) }
                ?: throw IllegalArgumentException("Marital status invalid: $value")
    }
}